velox/exec/tests/ExchangeFuzzer.cpp

```diff
@@ -0,0 +1,452 @@
+    LOG(INFO) << "Iteration " << iteration << " shuffl " << numSourceTasks
+              << "x" << numDestinationTasks << " drivers=" << numDriversPerTask
+              << " Type " << vectors.front()->type()->toString()
+              << " output buffer=" << outputBufferBytes
+              << " exchange buffer=" << exchangeBufferBytes
+              << " target batch=" << batchBytes;
+    configSettings_[core::QueryConfig::kMaxPartitionedOutputBufferSize] =
+        fmt::format("{}", outputBufferBytes);
+    configSettings_[core::QueryConfig::kMaxExchangeBufferSize] =
+        fmt::format("{}", exchangeBufferBytes);
+    configSettings_[core::QueryConfig::kPreferredOutputBatchBytes] =
+        fmt::format("{}", batchBytes);
+
+    auto& rowType = vectors.front()->type()->as<TypeKind::ROW>();
+    std::vector<std::string> aggregates;
+    std::vector<std::vector<TypePtr>> rawInputTypes;
+    for (auto i = 1; i < rowType.size(); ++i) {
+      rawInputTypes.push_back({rowType.childAt(i)});
+    }
+    auto expected = expectedChecksums(
+        vectors, rawInputTypes, numSourceTasks * numDriversPerTask);
+
+    std::vector<std::shared_ptr<Task>> tasks;
+
+    std::vector<std::string> leafTaskIds;
+    auto leafPlan = exec::test::PlanBuilder()
+                        .values(vectors, true)
+                        .partitionedOutput({"c0"}, numDestinationTasks)
+                        .planNode();
+
+    for (int32_t counter = 0; counter < numSourceTasks; ++counter) {
+      auto leafTaskId = makeTaskId(iteration, "leaf", counter);
+      leafTaskIds.push_back(leafTaskId);
+      auto leafTask = makeTask(leafTaskId, leafPlan, counter);
+      tasks.push_back(leafTask);
+      leafTask->start(numDriversPerTask);
+    }
+
+    std::vector<std::string> partialAggTaskIds;
+    auto partialAggPlan =
+        exec::test::PlanBuilder()
+            .exchange(leafPlan->outputType())
+            .partialAggregation({}, makeAggregates(rowType, 1))
+            .partitionedOutput({}, 1)
+            .planNode();
+
+    std::vector<exec::Split> partialAggSplits;
+    for (int i = 0; i < numDestinationTasks; i++) {
+      auto taskId = makeTaskId(iteration, "partial-agg", i);
+      partialAggSplits.push_back(
+          exec::Split(std::make_shared<exec::RemoteConnectorSplit>(taskId)));
+      auto task = makeTask(taskId, partialAggPlan, i);
+      tasks.push_back(task);
+      task->start(numDriversPerTask);
+      addRemoteSplits(task, leafTaskIds);
+    }
+
+    auto plan = exec::test::PlanBuilder()
+                    .exchange(partialAggPlan->outputType())
+                    .finalAggregation(
+                        {},
+                        makeAggregates(*partialAggPlan->outputType(), 0),
+                        rawInputTypes)
+                    .planNode();
+
+    try {
+      exec::test::AssertQueryBuilder(plan)
+          .splits(partialAggSplits)
+          .assertResults(expected);
+      if (FLAGS_inject_failure) {
```

Comment:
not sure if I follow this one. Is this flag just so we end up failing and saving the state? Could we just have a "--save_output" flag instead?